controller/src/main/java/io/pravega/controller/store/client/StoreClientFactory.java

```diff
@@ -102,14 +101,13 @@ public static CuratorFramework createZKClient(ZKClientConfig zkClientConfig, Sup
     }
 
     @VisibleForTesting
-    public static class ZKClientFactory implements ZookeeperFactory {
+    protected static class ZKClientFactory implements ZookeeperFactory {
```

Comment:
I was suggesting that we simply do not change the modifier and leave it package protected. My suggestion is that we change the signature back to:

```
static class ZKClientFactory implements ZookeeperFactory
```

Reply:
The idea behind setting `ZKClientFactory` as `protected` was to be able of creating a mock object extending `StoreClientFactory` that could close the ZK client right after the invocation to `newZookeeperClient`. However, I changed that and simplified the testing:
1) `StoreClientFactoryTest`: Emulates the change of a ZK connection parameter and checks that the client is correctly closed when invoking `newZookeeperClient`, and further requests to the ZK client return a `SessionExpiredException`.
2) `ZKControllerServiceMainTest`: Now, this test also checks that the Controller can restart with the ZK client being closed, which is what happens when any of the ZK parameters change and `newZookeeperClient` is invoked (we close the ZK client after injecting a session expiration from the test itself).